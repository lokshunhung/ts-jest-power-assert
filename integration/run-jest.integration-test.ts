import execa = require('execa');

function runJest() {
    jest.setTimeout(15000);
    return execa('jest', ['--config=jest.config.js', '--useStderr', '--no-watchman', '--no-cache', '--no-color'], {
        env: process.env,
        reject: false,
    });
}

function stripTimestamp(output: string) {
    return output.replace(/(?<![A-z])\d+(\.\d+)? m?s(?![A-z])/g, '__TIME_RELATED_OUTPUT_REMOVED__');
}

function removeTestTimestampPlaceholder(output: string) {
    return output.replace(/\(__TIME_RELATED_OUTPUT_REMOVED__\)/g, '');
}

test('runs jest', async () => {
    const { stdout, stderr } = await runJest();
    expect({
        stdout: removeTestTimestampPlaceholder(stripTimestamp(stdout)),
        stderr: removeTestTimestampPlaceholder(stripTimestamp(stderr)),
    }).toMatchSnapshot();
});
