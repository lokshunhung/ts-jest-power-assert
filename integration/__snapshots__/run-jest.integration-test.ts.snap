// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runs jest 1`] = `
Object {
  "stderr": "FAIL src/add.spec.ts
  add
    ✓ 1 + 1 = 2
    ✓ 20 + 10 = 30
    ✕ 2 + 3 = 4

  ● add › 2 + 3 = 4

    assert(received)

    Expected value to be equal to:
      true
    Received:
      false

    Message:
        # src/add.spec.ts:11
      
      assert((0, add_1.add)(a, b) === c)
             |   |     |    |  |  |   | 
             |   |     |    |  |  |   4 
             |   |     |    2  3  false 
             |   |     #function#       
             5   Object{add:#function#} 
      
      [number] c
      => 4
      [number] (0, add_1.add)(a, b)
      => 5

       9 |         \${2}  | \${3}  | \${4}
      10 |     \`('$a + $b = $c', ({ a, b, c }) => {
    > 11 |         assert(add(a, b) === c);
         |         ^
      12 |     });
      13 | });
      14 |

        
      at Decorator.Object.<anonymous>.Decorator._callFunc (node_modules/empower-core/lib/decorator.js:114:29)
      at src/add.spec.ts:11:9

Test Suites: 1 failed, 1 total
Tests:       1 failed, 2 passed, 3 total
Snapshots:   0 total
Time:        __TIME_RELATED_OUTPUT_REMOVED__
Ran all test suites.",
  "stdout": "",
}
`;
